Tools/autotest/bisect-helper.py

```diff
@@ -0,0 +1,171 @@
+            x = p.stdout.readline()
+            if len(x) == 0:
+                returncode = os.waitpid(p.pid, 0)
+                if returncode:
+                    break
+                    # select not available on Windows... probably...
+                time.sleep(0.1)
+                continue
+            self.program_output += x
+            x = x.rstrip()
+            print("%s: %s" % (prefix, x))
+        (_, status) = returncode
+        if status != 0:
+            self.progress("Process failed (%s)" %
+                          str(returncode))
+            raise subprocess.CalledProcessError(
+                returncode, cmd_list)
+
+    def build(self):
+        '''run ArduCopter build.  May exit with skip or fail'''
+        self.run_program("WAF-clean", ["./waf", "clean"])
+        cmd_configure = ["./waf", "configure"]
+        pieces = [shlex.split(x)
+                  for x in opts.waf_configure_args]
+        for piece in pieces:
+            cmd_configure.extend(piece)
+        self.run_program("WAF-configure", cmd_configure)
+        cmd_build = ["./waf", "build"]
+        pieces = [shlex.split(x)
+                  for x in opts.waf_build_args]
+        for piece in pieces:
+            cmd_build.extend(piece)
+        try:
+            self.run_program("WAF-build", cmd_build)
+        except subprocess.CalledProcessError as e:
+            # well, it definitely failed....
+            if opts.build_failure_string is not None:
+                if opts.build_failure_string in self.program_output:
+                    self.progress("Found relevant build failure")
+                    self.exit_fail()
+                # it failed, but not for the reason we're looking
+                # for...
+                self.exit_skip()
+            else:
+                self.exit_fail()
+
+
+class BisectBuild(Bisect):
+
+    def __init__(self, opts):
+        super(BisectBuild, self).__init__(opts)
+
+    def run(self):
+        self.build()  # may exit with skip or fail
+        self.exit_pass()
+
+
+class BisectCITest(Bisect):
+
+    def __init__(self, opts):
+        super(BisectCITest, self).__init__(opts)
+
+
+if __name__ == '__main__':
+
+    parser = optparse.OptionParser("bisect.py ")
+    parser.add_option("--build",
+                      action='store_true',
+                      default=False,
+                      help="Help bisect a build failure")
+    parser.add_option("--build-failure-string",
+                      type='string',
+                      default=None,
+                      help="If supplied, must be present in"
+                      "build output to count as a failure")
+
+    group_build = optparse.OptionGroup(parser, "Build options")
+    group_build.add_option("", "--waf-configure-arg",
+                           action="append",
+                           dest="waf_configure_args",
+                           type="string",
+                           default=["--board skyviper-v2450"],
```

Comment:
Is there any reason to default to SkyViper?

Reply:
No reason at all!

It's the one I was using for testing since that was the build that was broken at the time.

If you can think of a build more likely to break (and that people care about...), no worries replacing it.

Reply:
Well, I was think of this default being empty and it would just build SITL by default. But I don't care much, feel free to merge it!

Reply:
agree, using SITL by default is better